src/main/java/com/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,119 @@
+package com.onlinebookstore.service.impl;
+
+import com.onlinebookstore.dto.order.CreateOrderRequestDto;
+import com.onlinebookstore.dto.order.OrderDto;
+import com.onlinebookstore.dto.order.OrderItemDto;
+import com.onlinebookstore.dto.order.UpdateOrderRequestDto;
+import com.onlinebookstore.exception.DataProcessingException;
+import com.onlinebookstore.mapper.OrderItemMapper;
+import com.onlinebookstore.mapper.OrderMapper;
+import com.onlinebookstore.model.Order;
+import com.onlinebookstore.model.OrderItem;
+import com.onlinebookstore.model.RoleName;
+import com.onlinebookstore.model.ShoppingCart;
+import com.onlinebookstore.model.Status;
+import com.onlinebookstore.model.User;
+import com.onlinebookstore.repository.order.OrderItemRepository;
+import com.onlinebookstore.repository.order.OrderRepository;
+import com.onlinebookstore.service.OrderService;
+import com.onlinebookstore.service.ShoppingCartService;
+import com.onlinebookstore.service.UserService;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Transactional
+    @Override
+    public OrderDto save(CreateOrderRequestDto requestDto, String userName) {
+        ShoppingCart shoppingCart = shoppingCartService.getShoppingCart(userName);
+        if (shoppingCart.getCartItems().isEmpty()) {
+            throw new DataProcessingException("Can't place an order. No items found in the "
+                    + "shopping cart of the user: " + userName);
+        }
+        Order order = orderMapper.toOrderEntity(requestDto);
+        if (order.getShippingAddress().isBlank()) {
+            order.setShippingAddress(shoppingCart.getUser().getShippingAddress());
+        }
+        order.setUser(shoppingCart.getUser());
+        order.setStatus(Status.PENDING);
+        order.setOrderDate(LocalDateTime.now());
```

Comment:
if you implement comments in Order class, you won't need to set status and date here

Reply:
fixed this